src/main/scala/zio/prelude/Traversable.scala

```diff
@@ -189,6 +189,26 @@ trait Traversable[F[+_]] extends Covariant[F] {
   def product[A](fa: F[A])(implicit ev: Identity[Prod[A]]): A =
     foldMap(fa)(Prod[A])
 
+  /**
+   * Reduces the collection to a summary value using the associative operation,
+   * returning `None` if the collection is empty.
+   */
+  def reduceAssociative[A: Associative](fa: F[A]): Option[A] =
+    foldMap(fa)(a => Option(a))
+
+  /**
+   * Reduces the collection to a summary value using the idempotent operation,
+   * returning `None` if the collection is empty.
+   */
+  def reduceIdempotent[A: Idempotent: Equal](fa: F[A]): Option[A] =
```

Comment:
Shouldn't we actually use the fact that it is idempotent to optimize this?

Reply:
> use the fact that it is idempotent

And we do! :) Just have a look at the line below:
```scala
Idempotent[A].idempotent
```